### PR TITLE
Allow IFilterOptions.condition to be set to a function, as the documentation describes

### PR DESCRIPTION
**Pull request by Dretch**
case 2. Improvement to existing type definition.
- See the existing documentation above the `IFilterOptions.condition` field

### CODE REVIEW

**Comment by joeskeen** [comment]
:+1: LGTM

File: ui-grid/ui-grid-tests.ts [hunk 1]
```diff
@@ -41,6 +41,9 @@ columnDef.filter = {
     selectOptions: [{value: 4, label: 'test'}],
     disableCancelButton: false
 };
+columnDef.filter.condition = (searchTerm:string, cellValue:any, row:uiGrid.IGridRow, column:uiGrid.IGridColumn):boolean => {
```
Comment (btesser-r7) [inline ui-grid/ui-grid-tests.ts]: The rest of the file has a space between the : and the type.  Please make consistent...

```
columnDef.filter.condition = (searchTerm: string, cellValue: any, row: uiGrid.IGridRow, column: uiGrid.IGridColumn): boolean => {
```

File: ui-grid/ui-grid.d.ts [hunk 2]
```diff
@@ -3844,7 +3844,7 @@ declare module uiGrid {
          * or you can supply a custom filter function that gets passed the
          * following arguments: [searchTerm, cellValue, row, column].
          */
-        condition?: number;
+        condition?: number | ((searchTerm:string, cellValue:any, row:IGridRow, column:IGridColumn) => boolean);
```
Comment (btesser-r7) [inline ui-grid/ui-grid.d.ts]: I'd consider separating the function into its own interface, but that's not necessary.... also you have extraneous parens (remove) and you're missing the space after the colon in the types:

```
condition?: number | (searchTerm: string, cellValue: any, row: IGridRow, column: IGridColumn) => boolean;
```

**Comment by btesser-r7** [comment]
@joeskeen Any thoughts on my comments?

**Comment by joeskeen** [comment]
@btesser-r7 spot on... thanks for taking a closer look at it!

**Comment by btesser-r7** [comment]
@vvakame Why was this merged?  please revert.  there were issues that @joeskeen agreed had be changed (despite his previous :+1: )

**Comment by joeskeen** [comment]
@btesser-r7, @Dretch did make changes according to your feedback.  Is there anything else (other than the parenthesis that I mentioned [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8332#issuecomment-191822488)?

**Comment by btesser-r7** [comment]
Oh.  My bad.... Sorry 😬

**Pull request merged**